src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { KeyboardEvent, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo
+  handleDeleteTodo?: (id: number) => void,
+  handleTogglingTodo?: (todoId: number, todoStatus: boolean) => void,
+  loadingTodoIds: number[],
+  updateTitle?: (todoId: number, title: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleDeleteTodo = () => { },
+  loadingTodoIds,
```

Comment:
passing this array here is bad for optimization since its being changed all the time. instead you could pass a boolean representing if the item is being processed